### ISSUE TITLE
Dynamic latency improvements

### ISSUE THREAD

**Issue by gavv** [roc-streaming/roc-toolkit#712]
Follow-up for #688, tasks for future releases:

- [ ] Don't start session from fixed latency. Instead, start from min latency, wait until accumulated amount becomes larger than jitter or max latency, and then start session.

- [ ] Don't take into account short jitter spikes. Maybe use running percentile instead of running maximum.

**Comment by gavv** [comment]
@baranovmv I did a few benchmarks and have an interesting observation.

Please take a look at the `Jitter + Niq + Losses` plot in the end of reports in this benchmark: https://github.com/roc-streaming/qa/tree/main/reports/20250210_gh712_peak_jitter

(Most reports have this section, but not all of them)

For example, this one: https://github.com/roc-streaming/qa/blob/main/reports/20250210_gh712_peak_jitter/05--2ghz--no_load--low_res/README.md

![Image](https://github.com/user-attachments/assets/3fa07252-ca42-4062-8959-f3769d71704b)

This plot shows three metrics from the other sections combined together:

- network incoming queue (niq) spikes (sudden decreasements of NIQ len, in milliseconds)
- current jitter (measured from RTS)
- peak jitter (calculated from jitter)
- missing samples (unrecovered losses, how many samples were missing and replaced with zeros)

On the plot we can see that a few times the following happened:

- we got some unrecovered losses (red bar)
- we got high NIQ spike (yellow dot) that is above both jitter (blue line) and peak jitter (orange line)
- but we got no spike in jitter; it was still lower than current peak jitter, so there was no reaction to this

In other words, in this cases, we *had* information that current latency is not enough and should be increased, we we didn't use it. I guess on burst losses RTS alone is not enough to see spikes. NIQ have this information, on the other hand.

So the question is, do we want to take NIQ into account (somehow) when calculating jitter or target latency?

**Comment by gavv** [comment]
**WIP** benchmark for new jitter meter algorithm: https://github.com/roc-streaming/qa/tree/main/reports/20250214_gh712_jitter_sim

It includes current jitter meter implementation ported from C++ to python, and a few tests that get jitter and peak jitter from csv dumps (from [previous benchmark](https://github.com/roc-streaming/qa/tree/main/reports/20250210_gh712_peak_jitter)) and check that python port of jitter meter calculates identical peak jitter from jitter.

(On plots, jitter and peak jitter from dump are `rec_jitter` and `rec_peak_jitter`, and peak jitter calculated from `rec_jitter` is `sim_peak_jitter`. You can see that `rec_peak_jitter` and `sim_peak_jitter` do match on the plot).

Next step would be to test different algorithms for peak jitter calculation in python and compare them to the current one using these plots. Once we find a good one, it can be ported back to c++.
